### PR TITLE
Audit fixes final

### PR DESCRIPTION
**Pull request by ValliC444**
Made changes to fix some of the issues raised during audit - disabling autocomplete for login, changing delete user api to post, edit user access only for the creator.

### CODE REVIEW

File: Email/NMSReportingSuite/pom.xml [hunk 1]
```diff
@@ -143,34 +143,35 @@
...
             <artifactId>samples-spring-client</artifactId>
-            <version>LATEST</version>
+            <version>1.4.1</version>
         </dependency>
 
         <dependency>
             <groupId>org.apache.shiro</groupId>
             <artifactId>shiro-core</artifactId>
-            <version>LATEST</version>
+            <version>1.4.1</version>
```
Comment (anuranjan-alation) [inline Email/NMSReportingSuite/pom.xml:156]: Why not chnge in the parameter rather than hard-coding here?

File: NMSReportingSuite/pom.xml [hunk 2]
```diff
@@ -159,34 +159,35 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/ma
         </dependency>
 
         <!-- Apache Shiro -->
+<!--        latest versions no longer work with our projects, so switched to the lastest working version-->
         <dependency>
             <groupId>org.apache.shiro.samples</groupId>
             <artifactId>samples-spring-client</artifactId>
-            <version>LATEST</version>
```
Comment (anuranjan-alation) [inline NMSReportingSuite/pom.xml:165]: ditto

File: NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/controller/UserController.java [hunk 3]
```diff
@@ -510,12 +513,13 @@ public String forgotPassword(@RequestBody ForgotPasswordDto forgotPasswordDto, H
...
-
-    @RequestMapping(value = {"/deleteUser/{id}"}, method = RequestMethod.GET)
+//changed delete user to post, added a token verification
+    @RequestMapping(value = {"/deleteUser"}, method = RequestMethod.POST)
     @ResponseBody
-    public Map deleteExistingUser(@PathVariable("id") Integer id) {
+    public Map deleteExistingUser(HttpServletRequest request, @RequestBody Integer id) {
         User currentUser = userService.getCurrentUser();
-        if(currentUser != null){
+        String token = "dhty" + currentUser.getUserId().toString() + "alkihkf";
```
Comment (anuranjan-alation) [inline NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/controller/UserController.java:551]: is 'dhty' and 'alkijkf' random by choice? or can it be more meaningful?

File: app/views/login.html [hunk 4]
```diff
@@ -68,14 +68,15 @@
...
+					<input type="text" class="form-control input input-sm login-inputs" data-ng-model="user.username" name="username" placeholder="Username">
 
 				</div>
+<!--	changes for disbaling autocomplete for login-->
 				<div class="form-group" >
 					<span class="login-texts">Password</span>
-					<input name="password0" type="password"  value="Dummy" style="display: none" />
-					<input type="password" class="form-control password-input input input-sm login-inputs" id="divID" data-ng-model="user.password"  name="password" placeholder="Password"	 autocomplete="new-password" value="Dummy">
-					<input name="password1" type="password"  value="Dbmnmummy" style="display: none" />
+<!--					<input name="password0" type="password"  value="Dummy" style="display: none" />-->
```
Comment (anuranjan-alation) [inline app/views/login.html:77]: if not required, please delete the commented code. We can always retrieve these back from the history

File: NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/controller/UserController.java (hunk 3 above)
Comment (ValliC444) [reply NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/controller/UserController.java:551]: It's random by choice. They just wanted some token verification, so I am using a random string combined with userId

**Review by anuranjan-alation — APPROVED** [review]

**Review by anuranjan — APPROVED** [review]

**Pull request merged**